dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -578,18 +578,25 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = WorkflowVersion.class, responseContainer = "List")
     public Set<WorkflowVersion> requestDOIForWorkflowVersion(@ApiParam(hidden = true) @Auth User user,
         @ApiParam(value = "Workflow to modify.", required = true) @PathParam("workflowId") Long workflowId,
-        @ApiParam(value = "workflowVersionId", required = true) @PathParam("workflowVersionId") Long workflowVersionId) {
+        @ApiParam(value = "workflowVersionId", required = true) @PathParam("workflowVersionId") Long workflowVersionId,
+        @ApiParam(value = "This is here to appease Swagger. It requires PUT methods to have a body, even if it is empty. Please leave it empty.") String emptyBody) {
         Workflow workflow = workflowDAO.findById(workflowId);
         checkEntry(workflow);
         checkUser(user, workflow);
 
         WorkflowVersion workflowVersion = workflowVersionDAO.findById(workflowVersionId);
         if (workflowVersion == null) {
-            LOG.error(user.getUsername() + ": could not find version: " + workflow.getPath());
+            LOG.error(user.getUsername() + ": could not find version: " + workflow.getWorkflowPath());
             throw new CustomWebApplicationException("Version not found.", HttpStatus.SC_BAD_REQUEST);
 
         }
 
+        //Only issue doi if workflow is frozen.
+        if (!workflowVersion.isFrozen()) {
+            LOG.error(user.getUsername() + ": Could not generate DOI for " +  workflowNameAndVersion(workflow, workflowVersion) + ", frozen version required to generate DOI.");
+            throw new CustomWebApplicationException("Could not generate DOI for " + workflowNameAndVersion(workflow, workflowVersion) + ". Frozen version required to generate DOI.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Ideally, the `Frozen version required to generate DOI` segment should be a final static String and then used both here and in the test. That way, if we slightly modify the warning message, the test won't be automatically be broken.